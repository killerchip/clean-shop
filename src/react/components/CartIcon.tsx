import { View } from "react-native";
import { Badge, Icon } from "react-native-paper";

type Props = {
  itemsNumber?: number;
  size?: number;
  testID?: string;
};
export function CartIcon({ itemsNumber, size = 36, testID }: Props) {
  return (
    <View style={{ width: size, height: size }} testID={testID}>
      <Icon size={size} source="cart" />

      {itemsNumber !== undefined && itemsNumber > 0 && (
        <Badge style={{ position: "absolute" }} testID={`${testID}-cart-badge`}>
          {itemsNumber}
        </Badge>
      )}
    </View>
  );
}
