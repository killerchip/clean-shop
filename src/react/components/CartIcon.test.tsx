import { render, screen } from "@testing-library/react-native";
import { CartIcon } from "./CartIcon";
import "@testing-library/react-native/extend-expect";

describe("CartIcon Component", () => {
  it("renders correctly with default size", () => {
    render(<CartIcon testID="cart-icon" />);
    expect(screen.getByTestId("cart-icon")).toHaveStyle({
      width: 36,
      height: 36,
    });
  });

  it("renders correctly with custom size", () => {
    const customSize = 48;
    render(<CartIcon size={customSize} testID="cart-icon" />);
    expect(screen.getByTestId("cart-icon")).toHaveStyle({
      width: customSize,
      height: customSize,
    });
  });

  it("does not show badge when itemsNumber is 0", () => {
    render(<CartIcon itemsNumber={0} testID="cart-icon" />);
    expect(screen.queryByTestId("cart-icon-cart-badge")).toBeNull();
  });

  it("shows badge when itemsNumber is greater than 0", () => {
    const itemsNumber = 5;
    render(<CartIcon itemsNumber={itemsNumber} testID="cart-icon" />);
    expect(screen.getByTestId("cart-icon-cart-badge")).not.toBeNull();
    expect(screen.getByTestId("cart-icon-cart-badge")).toHaveTextContent(
      itemsNumber.toString(),
    );
  });

  it("does not show badge when itemsNumber is undefined", () => {
    render(<CartIcon testID="cart-icon" />);
    expect(screen.queryByTestId("cart-icon-cart-badge")).toBeNull();
  });

  it("matches snapshot", () => {
    render(<CartIcon testID="cart-icon" itemsNumber={5} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
