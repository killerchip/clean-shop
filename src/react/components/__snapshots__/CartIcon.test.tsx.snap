// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon Component matches snapshot 1`] = `
<View
  style={
    {
      "height": 36,
      "width": 36,
    }
  }
  testID="cart-icon"
>
  <Text
    accessibilityElementsHidden={true}
    allowFontScaling={false}
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    selectable={false}
    style={
      [
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 36,
        },
        [
          {
            "lineHeight": 36,
            "transform": [
              {
                "scaleX": 1,
              },
            ],
          },
          {
            "backgroundColor": "transparent",
          },
        ],
        {
          "fontFamily": "material-community",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    Û∞Ñê
  </Text>
  <Text
    collapsable={false}
    numberOfLines={1}
    style={
      {
        "alignSelf": "flex-end",
        "backgroundColor": "rgba(179, 38, 30, 1)",
        "borderRadius": 10,
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 10,
        "height": 20,
        "lineHeight": 10,
        "minWidth": 20,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 3,
        "position": "absolute",
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
    testID="cart-icon-cart-badge"
  >
    5
  </Text>
</View>
`;
